forwarder/writer.go

```diff
@@ -34,17 +35,27 @@ const (
 	subsystem = "forwarder"
 )
 
+var (
+	_ remote.WriteServer   = &Forwarder{}
```

Comment:
I think a better convention for ensuring that the Forwarder is a type of `remote.WriteServer` and `prometheus.Collector` is to just put those in the Forwarder struct definition.

e.g.

```golang
type Forwarder struct {
    remote.WriteServer
    prometheus.Collector

    writer bus.Writer
    ...
}
```